controllers/api/BaseApiController.php

```diff
@@ -148,4 +150,18 @@ private function eagerLoadRelations()
             $this->queryBuilder->with($this->eagerLoad);
         }
     }
+
+    private function applySorting()
+    {
+        $sorts = (array) Request::get('sort', null);
+
+        if (!empty($sorts) && !empty($sorts[0])) {
+            foreach ($sorts as $sort) {
+                $sortCriteria = explode(':', $sort);
```

Comment:
Try this:
```php
[$sortField, $sortOrder] = explode(':', $sortCriteria); // rename $sorts above to $sortCriterias and update the rest

if ($sortField) {
    $this->queryBuilder->orderBy($sortField, $sortOrder :? self::DEFAULT_SORT_ORDER);
}
```
And remove `DEFAULT_SORT_FIELD`.

Reply:
Changed. I made a bit different by using array_pad() with explode() to preserve default value for $sortOrder in case of the criteria string doesn't contain the order (like sort[]=position). If not use this, it will throw undefined index error.